### PR TITLE
src: maintainers: Refactoring, fixing bugs and adding verbose mode

### PR DESCRIPTION
**Pull request by aquilamacedo**
This pull request standardizes the file structure, fixe
a bug related to issue #932 and adds verbose mode.

### CODE REVIEW

File: src/maintainers.sh [hunk 1]
```diff
@@ -86,25 +51,25 @@ function maintainers_main()
...
   # if given path is not a patchfile, add -f to get_maintainer.pl options
   if ! is_a_patch "$FILE_OR_DIR"; then
-    if "$update_patch"; then
+    if [[ -n "$update_patch" ]]; then
       complain 'Option --update-patch was passed but given path is not a patch.'
       return 1
     fi
     is_file_a_patch=false
-    script_options="$script_options -f"
+    script_options="${script_options} -f"
```
Comment (rodrigosiqueira) [inline src/maintainers.sh:74]: Maybe check if there is a long version of `-f`.

File: src/maintainers.sh [hunk 2]
```diff
@@ -130,28 +95,63 @@ function maintainers_main()
   cd "$original_working_dir" || exit_msg 'It was not possible to move back from kernel dir'
 
   say "$SEPARATOR"
-  if "$update_patch"; then
+  if [[ -n $update_patch ]]; then
```
Comment (rodrigosiqueira) [inline src/maintainers.sh:114]: Add double quotes around this variable.

File: src/maintainers.sh [hunk 3]
```diff
@@ -130,28 +95,63 @@ function maintainers_main()
...
     print_files_authors "$FILE_OR_DIR"
   fi
 }
 
+# Prints the authors of a given file or files inside a given dir.
+#
+# @FILE_OR_DIR The argument is a file or directory path
+function print_files_authors()
+{
+  local FILE_OR_DIR="$1"
```
Comment (rodrigosiqueira) [inline src/maintainers.sh:146]: Add documentation for this first parameter.

File: src/maintainers.sh [hunk 4]
```diff
@@ -130,28 +95,63 @@ function maintainers_main()
...
+  elif [[ -f "$FILE_OR_DIR" ]]; then
+    files+=("$FILE_OR_DIR")
+  fi
+
+  local printed_authors_separator=false
+
+  for file in "${files[@]}"; do
+    authors=$(grep -oE 'MODULE_AUTHOR *\(.*\)' "$file" |
+      sed -E 's/(MODULE_AUTHOR *\( *\"|\" *\))//g' |
+      sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/, /g')
```
Comment (rodrigosiqueira) [inline src/maintainers.sh:163]: Use the long options for all of these commands.

File: src/maintainers.sh [hunk 5]
```diff
@@ -91,21 +95,26 @@ function maintainers_main()
...
       say "Maintainers already in 'To:' field of $(basename "${FILE_OR_DIR}")"
       return 0
-    elif grep -q -E '^To: ' "$FILE_OR_DIR"; then
-      # append maintainers to existing "To:" field
-      sed -E -i 's/(^To:.*)/\1, '"$script_output"'/' "$FILE_OR_DIR"
+    fi
+
+    cmd_manager "$flag" "grep --quiet --extended-regexp '^To: ' ${FILE_OR_DIR}"
+    if [[ "$?" == 0 ]]; then
+      cmd_manager "$flag" "sed -E -i 's/(^To:.*)/\1, ${script_output_copy}/' ${FILE_OR_DIR}"
```
Comment (rodrigosiqueira) [inline src/maintainers.sh:125]: use the long options.

File: src/maintainers.sh
Comment (rodrigosiqueira) [inline src/maintainers.sh:1]: Could you add a unit test for this?

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

File: src/maintainers.sh (hunk 1 above)
Comment (aquilamacedo) [reply src/maintainers.sh:74]: I didn't find the long option for this

File: src/maintainers.sh
Comment (aquilamacedo) [reply src/maintainers.sh:1]: sure, i'm working on it

File: src/maintainers.sh (hunk 1 above)
Comment (rodrigosiqueira) [reply src/maintainers.sh:74]: Np

**Comment by rodrigosiqueira** [comment]
Hi @aquilamacedo , overall, it looks good, but could you try to add some basic tests?

**Comment by rodrigosiqueira** [comment]
Based on our conversation, I decided to merge this PR.

**Pull request closed without merging**